reader/src/main/java/com/meituan/ptubes/reader/producer/mysqlreplicator/utils/MySQLUtils.java

```diff
@@ -206,33 +210,28 @@ public static Pair<Long, String> getMillisFromDatetime2(long value) {
 			final Calendar c = Calendar.getInstance();
 			c.set(year, month - 1, day, hour, minute, second);
 			c.set(Calendar.MILLISECOND, 0);
-			millisDateTime = new DateTime(c.getTimeInMillis());
+			validDateTime = new DateTime(c.getTimeInMillis());
+			invalidDateTime = String.format("%04d-%02d-%02d %02d:%02d:%02d", year, month, day, hour, minute, second);
 		}
-		stringDateTime = String.format("%04d-%02d-%02d %02d:%02d:%02d", year, month, day, hour, minute, second);
-		return Pair.of(millisDateTime.getMillis(), stringDateTime);
+		return Pair.of(validDateTime.getMillis(), invalidDateTime);

```

Comment:
How can sdk get the invalidDateTime from event?

Reply:
datetime2ToTimestamp will call it